### DISCUSSION TITLE
更新日志 - 22.03.5 - 20230811

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#947] [Announcements]
本次更新只包含x86以及rk35xx相关机型，不受相关功能影响的话可忽略此次更新

# 设备
* x86: 初始化时对于单网口设备默认开启LAN口的NAT，以支持 Docker 容器联网 #934
* H69K/H88K: 开机时执行5G模块的复位流程，以满足模块的时序要求

**Comment by gj731** [comment]
R5C，使用MT7921，wifi6模式几乎不可用，设备很难连接上，即便是连接上了，也非常不稳定，设置成wifi5模式比较稳定，算是解决了便携式设备的用网需求，但还想进一步提升，又买了一块MT7922，发现好像并没有MT7922的驱动程序，能否添加MT7922的驱动程序啊，谢谢垃。

**Reply by jjm2473** [reply]
看原版openwrt有没有，如果有就能加上，没有就不知道驱动在哪了

**Reply by gj731** [reply]
我是小白用户，不知道该怎么找或是怎么在咱的系统里边添加，不过我看有的系统里边集成软件包更新列表里边是有MT7922的，我不知道怎么提取安装。恳请您看看呗，毕竟现在的MTK确实挺好的，这个MT7921的体验真的非常好了，只是相对落后的协商速率还有目前体验下来只能开wifi5来使用。

**Reply by jjm2473** [reply]
所以，到底哪些固件有？我看openwrt下一个版本都还没

**Reply by jjm2473** [reply]
下载这两个文件：
* [WIFI_MT7922_patch_mcu_1_1_hdr.bin](https://github.com/openwrt/mt76/raw/d70546462b7b51ebc2bcdd5c534fdf3465be62a4/firmware/WIFI_MT7922_patch_mcu_1_1_hdr.bin)
* [WIFI_RAM_CODE_MT7922_1.bin](https://github.com/openwrt/mt76/raw/d70546462b7b51ebc2bcdd5c534fdf3465be62a4/firmware/WIFI_RAM_CODE_MT7922_1.bin)  

下载后放在 /lib/firmware/mediatek/ 里面，然后重启设备，看看 MT7922 能不能识别到

**Reply by gj731** [reply]
今天尝试了一下，MT7922在AC模式下，可以开出来160Mhz，但是设备看不到信号，也无法连接，AX模式下，无法开启160Mhz，只能开出来80Mhz，149-161信道都是正常可用的。。。MT7921在AC模式下都是正常的，但是在AX模式下，161信道开不出来，5.8G高频只有149和157两个信道正常。目前使用来看，MT7921和MT7922几乎没有区别，期待将来MT7922的驱动更加完善，能够正常开出来AX模式的160Mhz频宽，那样就跟MT7921能够拉开差距了。

**Reply by jjm2473** [reply]
看不到热点的话，试试选择国家，换信道，有些信道不是每个国家都支持，有些信道则需要规避雷达扫描（5.2-5.7Ghz需要一分钟后才能启动）。目前是同一个驱动，只是fw不一样。

**Comment by xixiluhui520** [comment]
老大，今天我的istoreos CPU使用率一直40几%居高不下，我该怎样去排除啊！

**Reply by jjm2473** [reply]
用htop命令看，默认按CPU使用率排序，如果不是你就点一下CPU。

**Reply by xixiluhui520** [reply]
感谢，知道了

**Comment by Jouleken** [comment]
n5105主机 esxi 虚拟 ikuai为主路由，iStoreOS为旁路由，
这版iStoreOS更新完之后，ikuai里面很多设备都不显示在线，端口映射也失效，iStoreOS回退到上一个版本后ikuai恢复正常。

**Reply by a447948814** [reply]
看看是不是设置问题，我更新有其他问题 ，重置设置，就好了

**Reply by jjm2473** [reply]
可能是因为NAT了，如果不需要NAT，可以自己在 iStoreOS 防火墙那边把lan区域的“IP动态伪装”关掉，但是这会导致 iStoreOS 里面的 Docker 无法联网。

**Reply by Jouleken** [reply]
感谢，按照这个方法，更新到最新版本的iStoreOS后，关闭LAN区域的“IP动态伪装”，内网设备 ikuai 端口映射正常。

**Reply by Jouleken** [reply]
有一个新问题：iStoreOS重启之后LAN区域的“IP动态伪装”会自动打开。

**Reply by nichwang88** [reply]
一样的问题，重启之后LAN区域的“IP动态伪装”会自动打开，每次都要手动关闭“IP动态伪装”，否则端口映射就会出问题。

**Reply by jjm2473** [reply]
不可能，除非你把/etc/config/network删了

**Reply by nichwang88** [reply]
确实这个问题存在，之前版本都没有发现端口映射异常，更新了这个版本以后发现这个问题是IP动态伪装默认打开导致的。
重启路由器确实会导致IP动态伪装再次开启。

**Reply by jjm2473** [reply]
单网口的才会默认打开，而且只在第一次启动打开，之后就不管了

**Reply by nichwang88** [reply]
我的就是单网口旁路由模式，目前状态就是每次重启就默认打开。

**Reply by jjm2473** [reply]
OK，那我之后弄个单网口的虚拟机测测

**Reply by jjm2473** [reply]
我用单网口虚拟机测试了，没有问题。第一次开机会默认打开NAT，手动取消以后重启并不会自动再打开。

**Reply by jjm2473** [reply]
我觉得你可以ikuai配置成旁路由，然后把ikuai的网关设置成istoreos，这样的话，ikuai就会转发所有的流量，也就知道客户端了。

**Reply by nichwang88** [reply]
我今天又确认了一次，重启依然会改为默认IP动态伪装模式。必须手动变更应用才能恢复正常。
我的主路由是ROS硬路由，istoreos是测试用的，只有部分设备通过该旁路由，还请再确认下到底是哪里出了问题，之前版本都正常的，就这个版本会出现这个问题。

**Reply by jjm2473** [reply]
没法复现啊，要不你恢复出厂设置试试

**Reply by nichwang88** [reply]
我已经回退上一个版本了，现在正常了。
等你下一个版本，如何问题依然存在，我再尝试重新恢复吧，主要是我自行安装了几个插件包括openclash、qemu-ga等，我怕还原出问题，这几天没空测试折腾。

**Reply by jjm2473** [reply]
你可以在升级以后删掉 /etc/board.d/99-default_network，就不可能再修改防火墙配置了。

**Reply by TommyLiu-1985** [reply]
我也是一摸一样的情况，重启依然会改为默认IP动态伪装模式，必须手动变更应用才能恢复正常。也退回旧版本了。

**Reply by jjm2473** [reply]
都说了删掉/etc/board.d/99-default_network就行了

**Reply by TommyLiu-1985** [reply]
![image](https://github.com/istoreos/istoreos/assets/82349643/0938778e-75f1-47b8-9593-7c688782d833)
删掉重启还是会改为默认IP动态伪装模式。

**Reply by jjm2473** [reply]
OK，那可能是在系统还没挂载overlay的时候就执行了99-default_network，不知道为啥虚拟机又不会这样。我之后加多一个判断条件，只在系统初始化后期才能执行。

**Reply by jjm2473** [reply]
刷这个测试固件看看， EFI版本 https://we.tl/t-v2QjzwgXKr ，非EFI版本 https://we.tl/t-v1Xjpx00vk

**Reply by jjm2473** [reply]
测试固件应该不删文件都行

**Reply by TommyLiu-1985** [reply]
我直接升级到了测试固件EFI版本。测试版已经没有问题了。

**Reply by jjm2473** [reply]
OK

**Comment by xixiluhui520** [comment]
老大istoreos可以安装1Panel管理面板吗？有没有办法安装呀！！！

**Reply by jjm2473** [reply]
什么东西啊，看看有没有docker版，用docker部署

**Comment by wjs876046992** [comment]
<img width="349" alt="image" src="https://github.com/istoreos/istoreos/assets/17695195/44c6b96e-c212-4c7d-a188-45182730d8b3">
<img width="300" alt="image" src="https://github.com/istoreos/istoreos/assets/17695195/6108b625-724e-4cea-abb1-762b5f7c86a7">
<img width="1477" alt="image" src="https://github.com/istoreos/istoreos/assets/17695195/29cc6b00-1bbd-4236-bf83-c98bff354446">

X86 更新之后，dns怎么设置都不对，只能手动调整/etc/resolv.conf，添加nameserver 114.114.114.114才能上网。怀疑是dnsmasq出了问题，但是不知道怎么处理，求教

**Reply by wjs876046992** [reply]
想不到对应的解决方法，就下载了个smartdns，然后就好了。

**Reply by jjm2473** [reply]
/etc/resolv.conf是一个软连接，指向/tmp/resove.conf，这个文件是自动生成的，不需要改，而且会被覆盖。如果有问题，先恢复出厂设置看看能不能复现。

**Reply by jjm2473** [reply]
可能就是因为你之前安装过什么插件导致dnsmasq的端口被改了

**Reply by wjs876046992** [reply]
应该不是的，我只是升级了下固件，然后提示固件写入失败，断电重启设备就这样了。

今天看到有固件升级提醒了。。。

**Reply by jjm2473** [reply]
应该不会有固件写入失败的提示，在刷固件的时候，web服务已经停止了，不可能返回结果给浏览器，浏览器只是会不断检查设备是否重新上线，如果上线了就跳转到新的页面。你应该看看系统日志有没有dnsmasq的报错。

**Reply by jjm2473** [reply]
要么就恢复出厂设置

**Reply by wjs876046992** [reply]
看了下日志，dnsmasq启动失败了，原因是MWAN3的问题，找了下是多线多拨插件，卸载MWAN3和Smart DNS后完全好了。
在我升级这个版本之前都是没问题的。

**Reply by jjm2473** [reply]
可能你是跨大版本升级，之前安装的插件不兼容了而已，有些插件重新安装以后就正常了。
